tests/unit/sources/test_base.py

```diff
@@ -69,8 +70,13 @@ class FaultySource(SourceHandler):
             FaultySource(  # type: ignore
                 source=Path(), part_src_dir=Path(), cache_dir=Path()
             )
-        assert str(raised.value) == (
-            "Can't instantiate abstract class FaultySource with abstract methods pull"
+        assert (
```

Comment:
The `pytest.raises` function has a `match` keyword exactly for this use. So, you could do:

```
expected = "^Can't instantiate abstract class FaultySource with abstract methods? pull$"
with pytest.raises(TypeError, match=expected) as raised:
    ...
```

Reply:
Nice! Thanks